File: src/main/kotlin/sigma/software/leovegas/drugstore/service/ProductServiceI.java
package sigma.software.leovegas.drugstore.service;

import java.util.List;
import sigma.software.leovegas.drugstore.dto.ProductDto;

public interface ProductServiceI {

    public List<ProductDto> getAll();

    public ProductDto getOne(Long id);

    public ProductDto create (ProductDto productDto);

    public ProductDto update (Long id, ProductDto productDto);

    public void delete (Long id);

}

File: src/main/kotlin/sigma/software/leovegas/drugstore/service/ProductServiceI.kt
package sigma.software.leovegas.drugstore.service

import sigma.software.leovegas.drugstore.dto.ProductRequest
import sigma.software.leovegas.drugstore.dto.ProductResponse

interface ProductServiceI {

    fun getAll(): MutableList<ProductResponse>

    fun save(product: ProductRequest): ProductResponse

    fun getOne(id: Long): ProductResponse

    fun update(id: Long, product: ProductRequest): ProductResponse

    fun delete(id: Long)

}
